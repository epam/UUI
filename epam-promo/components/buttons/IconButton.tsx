import { createSkinComponent } from '@epam/uui-core';
import { EpamPrimaryColor } from '../types';
import * as uui from '@epam/uui';

interface IconButtonMods {
    /**
     * Defines component color.
     * @default 'gray60'
     */
    color?: EpamPrimaryColor | 'gray30' | 'gray50' | 'gray60' | uui.IconButtonProps['color'];
}

/** Represents the properties for the IconButton component. */
export type IconButtonProps = uui.IconButtonCoreProps & IconButtonMods;

export const IconButton = /* @__PURE__ */createSkinComponent<uui.IconButtonProps, IconButtonProps>(
    uui.IconButton,
    (props) => ({ color: props.color ?? 'gray60' }),
);
