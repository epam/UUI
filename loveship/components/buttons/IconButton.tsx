import { IconButton as uuiIconButton, IconButtonProps as UuiIconButtonProps } from '@epam/uui';
import { devLogger, withMods } from '@epam/uui-core';
import { EpamAdditionalColor, EpamPrimaryColor } from '../types';

export interface IconButtonMods {
    /**
     * @default 'night600'
     */
    color?: EpamPrimaryColor | EpamAdditionalColor | 'white' | 'night200' | 'night300' | 'night400' | 'night500' | 'night600';
}

export type IconButtonProps = Omit<UuiIconButtonProps, 'color'> & IconButtonMods;

export const IconButton = withMods<Omit<UuiIconButtonProps, 'color'>, IconButtonMods>(
    uuiIconButton,
    () => [],
    (props) => {
        if (__DEV__) {
            devLogger.warnAboutDeprecatedPropValue<IconButtonProps, 'color'>({
                component: 'IconButton',
                propName: 'color',
                propValue: props.color,
                condition: () => ['night200', 'night300', 'night400'].indexOf(props.color) !== -1,
            });
        }
        return {
            color: props.color ?? 'night600',
        } as IconButtonProps;
    },
);
