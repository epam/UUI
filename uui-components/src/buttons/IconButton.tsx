import * as React from 'react';
import { ButtonBaseProps, ButtonBase } from './ButtonBase';
import { IconContainer } from '../layout';
import css from './Button.scss';

export interface IconButtonBaseProps extends ButtonBaseProps {}

export class IconButton extends ButtonBase<IconButtonBaseProps> {
    constructor(props: IconButtonBaseProps) {
        super(props);
    }

    getClassName() {
        return [css.container];
    }

    getChildren() {
        return <IconContainer key="icon" icon={this.props.icon} />;
    }
}
