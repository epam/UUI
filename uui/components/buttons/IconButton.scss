@use  '../../assets/styles/index' as *;

.root {
    fill: var(--uui-icon-btn);

    &:global(.-clickable) {
        &:hover, &:focus {
            fill: var(--uui-icon-btn-hover);
        }

        &:active {
            fill: var(--uui-icon-btn-active);
        }
    }

    &:global(.uui-disabled) {
        fill: var(--uui-icon-btn-disabled);

        &:global(.-clickable) {
            &:hover, &:focus {
                fill: var(--uui-icon-btn-disabled);
            }

            &:active {
                fill: var(--uui-icon-btn-disabled);
            }
        }
    }
}
