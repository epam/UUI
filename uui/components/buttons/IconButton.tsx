import { IconButtonBaseProps, IconButton as uuiIconButton } from '@epam/uui-components';
import { withMods } from '@epam/uui-core';
import css from './IconButton.scss';

export type IconColor = 'info' | 'success' | 'warning' | 'error' | 'secondary' | 'default';
export const allIconColors: IconColor[] = ['info', 'success', 'warning', 'error', 'secondary', 'default'];

export interface IconButtonMods {
    color?: IconColor;
}

export interface IconButtonProps extends IconButtonBaseProps, IconButtonMods {}

function applyIconButtonMods(mods: IconButtonProps & IconButtonMods) {
    return [
        `icon-button-${mods.color || 'default'}`,
        css.root,
    ];
}

export const IconButton = withMods<IconButtonProps, IconButtonMods>(uuiIconButton, applyIconButtonMods);
