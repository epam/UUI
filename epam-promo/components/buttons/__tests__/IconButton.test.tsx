import React from 'react';
import { IconButton } from '../IconButton';
import renderer from 'react-test-renderer';
import { ReactComponent as CalendarIcon } from '../../../icons/calendar-18.svg';

describe('IconButton', () => {
    it('should be rendered correctly', () => {
        const tree = renderer.create(<IconButton />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with props', () => {
        const tree = renderer.create(<IconButton color="blue" onClick={jest.fn} icon={CalendarIcon} isDisabled={false} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
