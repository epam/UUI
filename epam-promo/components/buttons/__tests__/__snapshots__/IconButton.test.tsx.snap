// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should be rendered correctly 1`] = `
<button
  className="container uui-button-box uui-enabled root icon-color-gray60"
  href={null}
  onClick={[Function]}
  role="button"
  tabIndex={-1}
  type="button"
>
  <div
    className="container uui-icon uui-enabled"
    style={Object {}}
  />
</button>
`;

exports[`IconButton should be rendered correctly with props 1`] = `
<button
  aria-disabled={false}
  className="container uui-button-box uui-enabled -clickable root icon-color-blue"
  href={null}
  onClick={[Function]}
  role="button"
  tabIndex={0}
  type="button"
>
  <div
    className="container uui-icon uui-enabled"
    style={Object {}}
  >
    <svg
      className=""
    />
  </div>
</button>
`;
